basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -201,7 +201,7 @@ public String createSocketBinding() throws CommandFailedException, IOException {
         String name = "UndertowSocketBinding_" + RandomStringUtils.randomAlphanumeric(6);
         try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
             client.apply(new AddSocketBinding.Builder(name)
-                    .port(ThreadLocalRandom.current().nextInt(10000, 19999))
+                    .port(AvailablePortFinder.getNextAvailable())
```

Comment:
The port number should be guaranteed already by `java.net.ServerSocket(0)` as it tries to open server socket on that port. If it doesn't have rights to do so, such port will not be selected.